File: TestLeetCode/TwoPointer/ContainerWithMostWater.cs
// https://leetcode.com/problems/container-with-most-water/description/

public class MaxAreaSolution {
    public int MaxArea(int[] height) {
        int left = 0, right = height.Length - 1;
        int res = 0;

        while (left < right) {
            int l = height[left];
            int r = height[right];
            int cur = (right - left) * Math.Min(l, r);
            res = Math.Max(cur, res);
            if (l > r) {
                right--;
            } else {
                left++;
            }
        }

        return res;
    }
}

File: python/two_pointer/container_with_most_water.py
class ContainerWithMostWater:
    def max_area(self, heights: list[int]) -> int:
        left, right, best = 0, len(heights) - 1, 0
        while left < right:
            best = max(best, (right - left) * min(heights[left], heights[right]))
            if heights[left] <= heights[right]: left += 1
            else: right -= 1
        return best
